### PR TITLE
CORE-10594 Renaming all 'httprpc' sub-packages across the project to 'rest'

### PR DESCRIPTION
**Pull request by ben-millar**
This PR is the final major code change in the RPC -> Rest renaming initiative ([CORE-7040](https://r3-cev.atlassian.net/browse/CORE-7040)) and satisfies [CORE-10594](https://r3-cev.atlassian.net/browse/CORE-10594). 

The only changes are to sub-directories named `rpc` or `httprpc` with each of them being renamed to `rest`, and all relevant import paths being updated accordingly.

[CORE-7040]: https://r3-cev.atlassian.net/browse/CORE-7040?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CORE-10594]: https://r3-cev.atlassian.net/browse/CORE-10594?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by vkolomeyko** [comment]
Looks fine, just needs merge conflict to be resolved.

**Review by vkolomeyko — APPROVED** [review]
LGTM

**Review by conal-smith — APPROVED** [review]
Just briefly checked this, looks ok to me

File: libs/rest/rest-client/src/main/kotlin/net/corda/httprpc/client/processing/BodyParametersResolver.kt [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
+import net.corda.rest.annotations.RestQueryParameter
+import net.corda.rest.annotations.ClientRequestBodyParameter
+import net.corda.rest.client.serialization.objectMapper
+import net.corda.rest.tools.annotations.extensions.name
 import net.corda.utilities.trace
 import org.slf4j.LoggerFactory
 import java.lang.reflect.Method
 
-private val log = LoggerFactory.getLogger("net.corda.httprpc.client.internal.processing.BodyParametersResolver.kt")
+private val log = LoggerFactory.getLogger("net.corda.rest.client.internal.processing.BodyParametersResolver.kt")
```
Comment (charlieR3) [inline libs/rest/rest-client/src/main/kotlin/net/corda/httprpc/client/processing/BodyParametersResolver.kt:12]: Looks like this doesn't line up with the package name
```suggestion
private val log = LoggerFactory.getLogger("net.corda.rest.client.processing.BodyParametersResolver.kt")
```

File: libs/rest/rest-client/src/main/kotlin/net/corda/httprpc/client/processing/FileParametersResolver.kt [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
+import net.corda.rest.tools.annotations.extensions.name
 import net.corda.utilities.trace
 import org.slf4j.LoggerFactory
 import java.lang.reflect.Method
 import java.lang.reflect.Parameter
-import net.corda.httprpc.HttpFileUpload
+import net.corda.rest.HttpFileUpload
 
-private val log = LoggerFactory.getLogger("net.corda.httprpc.client.internal.processing.FileParametersResolver.kt")
+private val log = LoggerFactory.getLogger("net.corda.rest.client.internal.processing.FileParametersResolver.kt")
```
Comment (charlieR3) [inline libs/rest/rest-client/src/main/kotlin/net/corda/httprpc/client/processing/FileParametersResolver.kt:12]:
```suggestion
private val log = LoggerFactory.getLogger("net.corda.rest.client.processing.FileParametersResolver.kt")
```

File: libs/rest/rest-client/src/main/kotlin/net/corda/httprpc/client/processing/FormParametersResolver.kt [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
-import net.corda.httprpc.tools.annotations.extensions.name
+import net.corda.rest.annotations.ClientRequestBodyParameter
+import net.corda.rest.tools.annotations.extensions.name
 import net.corda.utilities.trace
 import org.slf4j.LoggerFactory
 import java.lang.reflect.Method
 import java.lang.reflect.Parameter
 
-private val log = LoggerFactory.getLogger("net.corda.httprpc.client.internal.processing.FormParametersResolver.kt")
+private val log = LoggerFactory.getLogger("net.corda.rest.client.internal.processing.FormParametersResolver.kt")
```
Comment (charlieR3) [inline libs/rest/rest-client/src/main/kotlin/net/corda/httprpc/client/processing/FormParametersResolver.kt:10]:
```suggestion
private val log = LoggerFactory.getLogger("net.corda.rest.client.processing.FormParametersResolver.kt")
```

File: libs/rest/rest-client/src/main/kotlin/net/corda/httprpc/client/processing/BodyParametersResolver.kt (hunk 1 above)
Comment (charlieR3) [reply libs/rest/rest-client/src/main/kotlin/net/corda/httprpc/client/processing/BodyParametersResolver.kt:12]: (Wasn't added in your change though so I won't block on these comments)

File: libs/rest/rest-server-impl/src/test/kotlin/net/corda/rest/server/impl/apigen/processing/openapi/schema/SchemaModelToOpenApiSchemaConverterTest.kt [hunk 4]
```diff
@@ -20,7 +20,7 @@ internal class SchemaModelToOpenApiSchemaConverterTest {
             enum = listOf("A", "B", "C")
         )
 
-        val result = SchemaModelToOpenApiSchemaConverter.convert(model)
+        val result = net.corda.rest.server.impl.apigen.processing.openapi.schema.SchemaModelToOpenApiSchemaConverter.convert(model)
```
Comment (charlieR3) [inline libs/rest/rest-server-impl/src/test/kotlin/net/corda/rest/server/impl/apigen/processing/openapi/schema/SchemaModelToOpenApiSchemaConverterTest.kt:23]: Is this change necessary? Seems like they are in the same package. Same in the rest of the file.

File: components/membership/membership-rest-impl/src/main/kotlin/net/corda/membership/impl/rest/v1/RestInvokationUtils.kt [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+ *      prevented the future from completing).
+ *  - Throw [InternalServerException] if none of the above rules match.
+ *
+ *  @param logger Logger instance to register exceptions, if any, under WARN log level.
+ *  @param operation String representing the operation being executed.
+ *  @param ignoredExceptions List of exception types to ignore.
+ *  @param block Block to execute.
+ */
+@Suppress("ThrowsCount")
+internal fun <T> tryWithExceptionHandling(
```
Comment (charlieR3) [inline components/membership/membership-rest-impl/src/main/kotlin/net/corda/membership/impl/rest/v1/RestInvokationUtils.kt:22]: I think this file was deleted and the contents moved to `net.corda.httprpc.messagebus.MessageBusUtils` so this PR is reintroducing it

**Review by charlieR3 — COMMENTED** [review]
Added a couple comments. Can approve if you want to merge and follow up later since it's such a large PR. 
I think it's intentional, but a lot of classes had their package definition updated but they are still in a `httprpc` directory.

File: libs/rest/rest-server-impl/src/test/kotlin/net/corda/rest/server/impl/apigen/processing/openapi/schema/SchemaModelToOpenApiSchemaConverterTest.kt (hunk 4 above)
Comment (ben-millar) [reply libs/rest/rest-server-impl/src/test/kotlin/net/corda/rest/server/impl/apigen/processing/openapi/schema/SchemaModelToOpenApiSchemaConverterTest.kt:23]: Well spotted! This was IntelliJ trying to be "helpful" during the refactor; I nuked the rest of these such suggestions, but that one slipped through. Thank you!

**Review by charlieR3 — APPROVED** [review]
Spoke offline. Comments will be addressed later so approving this PR now.

**Comment by ben-millar** [comment]
Thanks @charlieR3 for the in-depth review, really good catches. As discussed offline, I'm gonna get this PR through now and then open a PR to address the comments right away.
